src/System.Management.Automation/engine/GetCommandCommand.cs

```diff
@@ -531,6 +531,17 @@ private void OutputResultsHelper(IEnumerable<CommandInfo> results)
                         {
                             PSObject syntax = PSObject.AsPSObject(result.Syntax);
 
+                            if (!_nameContainsWildcard && !this.Name[0].Equals(result.Name))
+                            {
+                                string replacedSyntax = string.Format(
+                                    "{0} -> {1}{2}{3}",
+                                    this.Name[0],
+                                    result is ExternalScriptInfo extScript ? string.Format("{0}{1}", extScript.Path, Environment.NewLine) : result.Name,
+                                    Environment.NewLine,
+                                    result.Syntax.Replace(result.Name, this.Name[0]));
```

Comment:
And how does output look with the replace?

Reply:
Also I see you checking for wildcards. Is the logic smart enough to show the alias properly if a wildcard search finds commands that are aliases and outputs the syntax for them? There should probably be a specific test just for that scenario to make sure it is covered.

Reply:
For scripts it looks like this:

```
PS> gcm tgs -syn
tgs -> C:\Users\Chris\Test-GcmSyntax.ps1

tgs [-Name] <string[]> [<CommonParameters>]
```

For commands it's:

```
PS>  gcm del -syn
del -> Remove-Item

del [-Path] <string[]> [-Filter <string>] [-Include <string[]>] [-Exclude <string[]>] [-Recurse] [-Force] [-Credential <pscredential>] 
[-WhatIf] [-Confirm] [-Stream <string[]>] [<CommonParameters>]

del -LiteralPath <string[]> [-Filter <string>] [-Include <string[]>] [-Exclude <string[]>] [-Recurse] [-Force] [-Credential <pscredential>] [-WhatIf] [-Confirm] [-Stream <string[]>] [<CommonParameters>]
```

Reply:
> Also I see you checking for wildcards. Is the logic smart enough to show the alias properly if a wildcard search finds commands that are aliases and outputs the syntax for them? There should probably be a specific test just for that scenario to make sure it is covered.

Not right now, I need to add something in for that.

With the codefactor tests complaining about it being too complex I'll break out this into it's own method as it'll need to handle this case too and that'll make it more difficult to follow I think.

Reply:
I don't like last commits. I think it is a task for Get-Help to return enhanced information in human format.

Reply:
@iSazonov Actually, it's a task for `Get-Command`, and that's exactly what it does. `Get-Command gci` shows you that `gci` is an alias for `Get-Command`. With this PR, `Get-Command gci -Syntax` shows you the syntax for the `gci` command and also identifies that it's an alias for `Get-ChildItem`. There's no reason to force users to have to call `Get-Command` twice to get such important information.

In fact, it would probably be far more useful for a less experienced user if `-Syntax` reported this additional information, and if the user had `$PSDefaultParameterValues['Get-Command:Syntax'] = $true` in their profile, because unless they are programmatically working with command metadata or trying to find out what module a command comes from, this information is far more useful for them.

Reply:
> There's no reason to force users to have to call Get-Command twice to get such important information.

Get-Help is for users to get all in once. Get-Command is more for scripting - without -Syntax it returns typed object (AliasInfo or CommandInfo) and I'd even expect that -Syntax returns typed output too as a syntax tree.